### DISCUSSION TITLE
Setup Notification for Empty water tank

### DISCUSSION THREAD

**Discussion by Sagi-bb** [humbertogontijo/homeassistant-roborock#418] [Q&A] [answered]
HI,
I have the Roborock S7 maxV ultra. 
I would like to received a notification via HA when the water tank is empty or the dirty water tank is full.
Can you help with the trigger for this automation ?

**Comment by humbertogontijo** [comment]
You mean the water tank from the dock, right? We are currently trying to discovery the commands for them. But both me and @Lash-L don't have an empty wash fill dock to verify these commands(to add sensors for them).

**Reply by Sagi-bb** [reply]
Yes from the dock.

**Reply by the-ress** [reply]
I could try to discover that on mine - do you have any pointers on where/how to start? Does I need to sniff traffic from the official app?

**Reply by humbertogontijo** [reply]
@the-ress Yes, you'll need to. Unless there are more fields on your response from get_status command(All fields are added as attributes for the vacuum)

**Reply by the-ress** [reply]
OK, I’ll try to do that this week. It’s not in the attributes, I already checked.

**Comment by the-ress** [comment]
I found out it was `dock_error_status` after all and the values are even already [in the library](https://github.com/humbertogontijo/python-roborock/blob/main/roborock/code_mappings.py#L225-L226). Also I found a new code for 'dust bin not installed' (46).

Here's what I saw:
* clean water empty - 38
* waste water full - 39
* both - still 38
* dust bin not installed - 46

**Reply by Sagi-bb** [reply]
![image](https://github.com/humbertogontijo/homeassistant-roborock/assets/68748153/d3609dcb-e29a-4ef1-b522-c51fe8008595)
 only found above attribute with the word duck

**Reply by the-ress** [reply]
cc @humbertogontijo

**Reply by the-ress** [reply]
edit: it is an entity, see post below

@Sagi-bb ~currently it's just an attribute of the vacuum entity, not its own entity~
![image](https://github.com/humbertogontijo/homeassistant-roborock/assets/1612377/6b1a7377-2879-4bfd-8651-ef5c0a8941a8)

![image](https://github.com/humbertogontijo/homeassistant-roborock/assets/1612377/e607f480-18f4-4063-bfd8-19e4fccbe5e3)

**Reply by Sagi-bb** [reply]
![image](https://github.com/humbertogontijo/homeassistant-roborock/assets/68748153/4712bd7b-eccd-4a81-a188-ea57ce5aa9bb)

so is this a good trigger

**Reply by humbertogontijo** [reply]
You can use the sensor sensor.<device_name>_dock_status. It has the same information as this attribute, but non numeric. I added the code 46 to the api, but it will require a new release of the integration to show it

**Reply by the-ress** [reply]
I didn't notice it was already an entity, thanks

@Sagi-bb when testing it, keep in mind, the error only appears when the vacuum actually needs water - when washing the mop etc.
(it's the same behavior as in the official app)

**Reply by Sagi-bb** [reply]
OK i will try it , do you know the different sensor status (in the drop down menu i can only see available / not available) ?

**Reply by Sagi-bb** [reply]
![image](https://github.com/humbertogontijo/homeassistant-roborock/assets/68748153/f16abc43-5c27-469e-a745-aa2fe8285e96)
sorry options are : unavailable/unknown

**Reply by humbertogontijo** [reply, marked as answer]
The options are the keys from [this enum](https://github.com/humbertogontijo/python-roborock/blob/2ea25d32bcd3c393d3585f17278a3523bf536abf/roborock/code_mappings.py#L220)

**Reply by Sagi-bb** [reply]
thank you i will try this

**Reply by Sagi-bb** [reply]
working thank you

**Reply by poudenes** [reply]
Those errors are not changing in HA for Roborock S8. Removed the dust bin and water tank but bro difference.

**Reply by the-ress** [reply]
For clean water tank and waste water tank, it’s only detected when the vacuum actually needs water (at least on S7)

**Reply by arne182** [reply]
In the app for the S8 it is shown immediately when the tanks are removed

**Reply by 7leman7** [reply]
That's what I'm seeing on S8 Pro Ultra as well - the app already shows clean water shortage, but dock status sensor = OK and dockerrorstatus = 0 in Home Assistant. The app seems to use another source for its dock status messages.

**Comment by poudenes** [comment]
Hi All,

I have found the follow status changed for my S8:
waterBoxStatus: 1 = Box is inserted
waterShortageStatus: 1 = Water is empty (also message in app)

After removing the box:
waterBoxStatus: 0 = Box is removed from S8
waterShortageStatus: 0 = Water status is ok (box still removed from S8)

After inserting box filled with water:
waterBoxStatus: 1 = Box is inserted again in S8
waterShortageStatus: 0 = Water status is ok
